utils/misc.ml

```diff
@@ -531,6 +531,16 @@ let create_hashtable size init =
   List.iter (fun (key, data) -> Hashtbl.add tbl key data) init;
   tbl
 
+let hash_seed =
+  let seed = Random.bits () in
+  if seed mod 2 = 0 then seed + 1 else seed
+
+let hash2 a b =
+  let a = Hashtbl.hash a in
+  let b = Hashtbl.hash b in
+  let r = a * hash_seed + b in
+  r lxor (r lsr 17)
```

Comment:
Could you include a rationale for using this particular hash algorithm ? In particular it seems that this algorithm could create hashes that do not fit in the 30 bit range that `Hashtbl.hash` returns, and even return negative hashes (I'm not sure whether this is a problem or not in practice).

Reply:
Ah, I asked Stephen Dolan and he said it was a good fast one.  I will check with him about the 30-bit thing; I think it's probably ok.

Reply:
I talked to Stephen and he agrees this is ok.